addons/main/functions/fnc_doCallout.sqf

```diff
@@ -74,15 +73,33 @@ if (isNil "_cachedSounds") then {
     _cachedSounds = getArray (_protocolConfig >> _callout);
     private _deleted = false;
     {
-        private _sound = _x;
+        private _sound = toLowerANSI _x;
         if (_sound == "") then {
-            _sound = objNull;
             _deleted = true;
-        } else {
-            if (_sound select [0, 1] != "\") then {
-                _sound = (getArray (configFile >> "CfgVoice" >> _speaker >> "directories") select 0) + _sound;
-            };
+            _cachedSounds set [_forEachIndex, objNull];
+            continue;
         };
+        
+        private _hasFileEnding = _sound regexMatch ".+?\.(?:ogg|wss|wav|mp3)$/io";
+
+        if (_sound select [0, 1] != "\") then {
+            _sound = (getArray (configFile >> "CfgVoice" >> _speaker >> "directories") select 0) + _sound;
+        };
+
+        if (!fileExists _sound && !_hasFileEnding) then {
```

Comment:
Switch condition order and add lazy eval for an easy optimization, ``fileExists`` is a lot slower than checking the variable

``if (!_hasFileEnding && {!fileExists _sound}) ..``

Reply:
good catch